plugins/module_utils/network/awplus/facts/class_maps/class_maps.py

```diff
@@ -0,0 +1,141 @@
+#
+# -*- coding: utf-8 -*-
+# Copyright 2023 Allied Telesis
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The awplus class_maps fact class
+It is in this file the configuration is collected from the device
+for a given resource, parsed, and the facts tree is populated
+based on the configuration.
+"""
+import re
+from copy import deepcopy
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common import (
+    utils,
+)
+from ansible_collections.alliedtelesis.awplus.plugins.module_utils.network.awplus.argspec.class_maps.class_maps import Class_mapsArgs
+
+
+class Class_mapsFacts(object):
+    """ The awplus class_maps fact class
+    """
+
+    def __init__(self, module, subspec='config', options='options'):
+        self._module = module
+        self.argument_spec = Class_mapsArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    @staticmethod
+    def get_class_map_conf(connection):
+        return connection.get("show running-config | begin class")
```

Comment:
This is not enough:

mls qos enable
mls qos map premark-dscp 23 to new-dscp 24
mls qos map premark-dscp 23 to new-bandwidth-**class** yellow
mls qos map premark-dscp 33 to new-dscp 34
mls qos map premark-dscp 33 to new-bandwidth-**class** red
!
class-map afd
!

Reply:
Line 87 does check that the configuration is a class-map before phasing data. 
I can change the way that class-maps are obtained quite easily without much issue, if you would like.